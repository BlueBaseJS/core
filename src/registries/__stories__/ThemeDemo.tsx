import { BlueBase, BlueBaseContext } from '../../';
import React from 'react';
import { Theme } from '../ThemeRegistry';
import { ThemeContext } from '../../themes';

export interface ThemeDemoProps {
	children?: React.ReactNode;
}

export class ThemeDemo extends React.PureComponent<ThemeDemoProps> {

	static contextType = BlueBaseContext;

	render() {
		const BB: BlueBase = (this as any).context;
		const { children } = this.props;

		return (
			<ThemeContext.Consumer children={({ theme }: { theme: Theme }) => {

				const mainStyle = {
					backgroundColor: theme.palette.background.default,
					padding: theme.spacing.unit
				};

				return (
					<BB.Components.View style={mainStyle}>
						<BB.Components.View>
							<BB.Components.Text style={theme.typography.body1}>
								{theme.name}
							</BB.Components.Text>
							{children}
						</BB.Components.View>
					</BB.Components.View>
				);
			}} />
		);
	}
}
