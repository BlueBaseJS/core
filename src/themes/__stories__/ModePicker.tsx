import { Picker } from 'react-native';
import React from 'react';
import { useTheme } from '../..';

export const ModePicker = () => {
	const { theme, changeMode } = useTheme();

	return (
		<Picker selectedValue={theme.mode} style={{ width: 150 }} onValueChange={changeMode}>
			<Picker.Item label="Light" value="light" />
			<Picker.Item label="Dark" value="dark" />
		</Picker>
	);
};
