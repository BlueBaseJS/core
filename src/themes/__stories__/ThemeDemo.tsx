import { BlueBase, BlueBaseContext } from '../../';
import React from 'react';
import { Theme } from '../../registries';
import { ThemeContext } from '../../themes';

export interface ThemeDemoProps {
	children?: React.ReactNode;
}

export class ThemeDemo extends React.PureComponent<ThemeDemoProps> {

	static contextType = BlueBaseContext;

	render() {
		const BB: BlueBase = (this as any).context;
		const { children } = this.props;

		return (
			<ThemeContext.Consumer children={({ theme }: { theme: Theme }) => {

				const mainStyle = {
					backgroundColor: theme.palette.background.default,
					padding: theme.spacing.unit
				};

				return (
					<BB.Components.View style={mainStyle}>
						<BB.Components.Body1>
							{theme.name}
						</BB.Components.Body1>
						{children}
					</BB.Components.View>
				);
			}} />
		);
	}
}
