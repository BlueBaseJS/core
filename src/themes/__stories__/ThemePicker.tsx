import { BlueBase, BlueBaseContext } from '../../';
import { ThemeContext, ThemeContextData } from '../../themes';
import { Picker } from 'react-native';
import React from 'react';

export class ThemePicker extends React.PureComponent {

	static contextType = BlueBaseContext;

	render() {
		const BB: BlueBase = (this as any).context;
		const themes = [...BB.Themes.entries()];
		return (
			<ThemeContext.Consumer children={({ changeTheme }: ThemeContextData) => (
				<Picker
					selectedValue={BB.Configs.getValue('theme.name')}
					style={{ width: 150 }}
					onValueChange={changeTheme}>
					{themes.map(entry => <Picker.Item label={entry[1].name} value={entry[0]} key={entry[0]} />)}
				</Picker>
			)} />
		);
	}
}
