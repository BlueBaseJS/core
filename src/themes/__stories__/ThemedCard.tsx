import { BlueBase, BlueBaseContext } from '../..';
import { TextStyle, ViewStyle } from 'react-native';
import { MaybeThunk } from '../../utils';
import React from 'react';
import { Theme } from '../../registries/ThemeRegistry';

export interface ThemedCardProps {
	styles?: {
		root: ViewStyle,
		text: TextStyle,
	}
}

export class ThemedCard extends React.PureComponent<ThemedCardProps> {

	static contextType = BlueBaseContext;

	static defaultStyles: MaybeThunk<{ root: ViewStyle, text: TextStyle }> = (theme: Theme) => ({
		root: {
			backgroundColor: theme.palette.background.card,
			borderRadius: theme.shape.borderRadius,
			height: 150,
			justifyContent: 'center',
			padding: theme.spacing.unit,
			width: 150,
		},

		text: {
			color: theme.palette.text.primary,
			textAlign: 'center',
		}
	})

	render() {
		const BB: BlueBase = (this as any).context;
		const { styles, children } = this.props;
		return (
			<BB.Components.View style={styles && styles.root}>
				<BB.Components.Text style={styles && styles.text}>
					{children}
				</BB.Components.Text>
			</BB.Components.View>
		);
	}
}
