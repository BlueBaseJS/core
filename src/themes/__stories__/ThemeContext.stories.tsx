// tslint:disable:no-console
import { BlueBase, BlueBaseConsumer } from '../..';
import React from 'react';
import { ThemeDemo } from '../../themes/__stories__/ThemeDemo';
import { ThemePicker } from '../../themes/__stories__/ThemePicker';
import { ThemeProvider } from '..';
import storiesOf from '@bluebase/storybook-addon';

storiesOf('ThemeContext', module)

.add('Basic Example', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.View>
			<ThemePicker />
			<ThemeDemo />
		</BB.Components.View>
	)} />
))

.add('Basic Example with overrides', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<ThemeProvider theme="bluebase-dark" overrides={{ palette: { background: { default: 'red' } } }}>
			<BB.Components.View style={{ margin: 8 }}>
				<ThemeDemo />
			</BB.Components.View>
		</ThemeProvider>
	)} />
))

.add('Nested Theme', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.View>
			<ThemePicker />
			<ThemeDemo>
				<ThemeProvider theme="bluebase-dark">
					<BB.Components.View style={{ margin: 8 }}>
						<ThemeDemo />
					</BB.Components.View>
				</ThemeProvider>
			</ThemeDemo>
		</BB.Components.View>
	)} />
))

.add('3 Level Nesting', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.View>
			<ThemePicker />
			<ThemeDemo>
				<ThemeProvider theme="bluebase-dark">
					<BB.Components.View style={{ margin: 8 }}>
					<ThemeDemo>
						<ThemeProvider>
							<BB.Components.View style={{ margin: 8 }}>
								<ThemePicker />
								<ThemeDemo />
							</BB.Components.View>
						</ThemeProvider>
					</ThemeDemo>
					</BB.Components.View>
				</ThemeProvider>
			</ThemeDemo>
		</BB.Components.View>
	)} />
));

storiesOf('Themes', module)

.add('A component with defaultStyles', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.View>
			<BB.Components.ThemedCard>
				A card with defaultStyles
			</BB.Components.ThemedCard>
			<ThemePicker />
		</BB.Components.View>
	)} />
))

.add('A component with styles from theme', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.ThemedCard2>
			A Card with styles from the theme (Color: Green)
		</BB.Components.ThemedCard2>
	)} />
))

.add('A component with styles from prop', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.ThemedCard2 styles={{ root: { backgroundColor: 'blue' } }}>
			A Card with styles from the theme (Color: Blue)
		</BB.Components.ThemedCard2>
	)} />
))

.add('A component with styles from component registry', () => (
	<BlueBaseConsumer children={(BB: BlueBase) => (
		<BB.Components.ThemedCard3>
			A Card with styles from component registry (Color: Orange)
		</BB.Components.ThemedCard3>
	)} />
));
