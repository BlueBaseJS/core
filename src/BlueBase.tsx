import { Analytics, Logger } from './api';
import {
	ComponentCollection,
	ComponentRegistry,
	ConfigCollection,
	ConfigRegistry,
	HookNestedCollection,
	HookRegistry,
	PluginCollection,
	PluginRegistry,
	ThemeCollection,
	ThemeRegistry,
} from './registries';
import { BlueBaseProvider } from './Context';
import React from 'react';
import { ThemeProvider } from './themes';
import { getComponent } from './getComponent';
import systemHooks from './hooks';

export interface BootOptions {
	components: ComponentCollection,
	configs: ConfigCollection,
	hooks: HookNestedCollection,
	plugins: PluginCollection,
	themes: ThemeCollection,
	// routes: Plugin[]
}

export class BlueBase {

	// APIs
	public Analytics = new Analytics(this);
	public Logger = new Logger(this);

	// Registries
	public Components = new ComponentRegistry(this);
	public Configs = new ConfigRegistry(this);
	public Hooks = new HookRegistry(this);
	public Plugins = new PluginRegistry(this);
	public Themes = new ThemeRegistry(this);

	// Flags
	public booted = false;

	private bootOptions: BootOptions = {
		components: {},
		configs: {},
		hooks: {},
		plugins: [],
		themes: [],
	};

	public async boot(options?: Partial<BootOptions> & { children?: React.ReactNode }) {

		this.bootOptions = { ...this.bootOptions, ...options };

		await this.Hooks.registerNestedCollection(systemHooks);
		await this.Hooks.registerNestedCollection(this.bootOptions.hooks);

		await this.Hooks.run('bluebase.boot', this.bootOptions);

		const SystemApp = getComponent('SystemApp');

		// Set View
		// const SystemApp = this.Components.resolve('SystemApp');
		// SystemApp = await this.Hooks.run('bluebase.system.app', SystemApp);

		const BlueBaseRoot = () => (
			<BlueBaseProvider value={this}>
				<ThemeProvider>
					<SystemApp>
						{options && options.children}
					</SystemApp>
				</ThemeProvider>
			</BlueBaseProvider>
		);

		this.booted = true;

		return BlueBaseRoot;
	}
}