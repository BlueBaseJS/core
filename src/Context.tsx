import { BlueBase } from './BlueBase';
import { createContext } from 'react';

export const BlueBaseContext: React.Context<BlueBase> = createContext(undefined as any);
export const BlueBaseProvider = BlueBaseContext.Provider;
export const BlueBaseConsumer = BlueBaseContext.Consumer;
