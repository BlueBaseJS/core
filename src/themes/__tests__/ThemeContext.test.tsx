import { BlueBase } from '../../BlueBase';
import { BlueBaseApp } from '../../index';
import { BlueBaseContext } from '../../Context';
import React from 'react';
import { ThemeDemo } from '../__stories__/ThemeDemo';
import { ThemePicker } from '../__stories__/ThemePicker';
import { ThemeProvider } from '../..';
import { mount } from 'enzyme';
import { waitForElement } from 'enzyme-async-helpers';

describe('ThemeContext', () => {
	test(`should render a ThemeDemo component with themed background color`, async () => {
		const BB = new BlueBase();
		await BB.boot();

		const wrapper = mount(
			<BlueBaseContext.Provider value={BB}>
				<ThemeProvider>
					<ThemeDemo />
				</ThemeProvider>
			</BlueBaseContext.Provider>
		);

		// Wait for render
		await waitForElement(wrapper as any, ThemeDemo);

		const view = wrapper
			.find('ThemeDemo')
			.childAt(0)
			.first();
		expect((view.prop('style') as any).backgroundColor).toBe('#f5f5f5');
	});

	test(`should change theme based on callback function`, async () => {
		const wrapper = mount(
			<BlueBaseApp>
				<ThemePicker />
				<ThemeDemo />
			</BlueBaseApp>
		);

		// Wait for render
		await waitForElement(wrapper, ThemeDemo);
		// expect(wrapper).toMatchSnapshot();

		// Check theme
		let view = wrapper
			.find('ThemeDemo')
			.childAt(0)
			.first();
		expect((view.prop('style') as any).backgroundColor).toBe('#f5f5f5');
		expect(
			wrapper
				.find('ThemeDemo')
				.find('Text')
				.last()
				.text()
		).toBe('BlueBase Light');

		// Change theme
		const onValueChange: any = wrapper.find('Picker').prop('onValueChange');
		onValueChange('bluebase-dark');

		await waitForElement(wrapper as any, ThemeDemo);
		// expect(wrapper).toMatchSnapshot();

		// Verify that background color is dark
		view = wrapper
			.find('ThemeDemo')
			.childAt(0)
			.first();
		expect((view.prop('style') as any).backgroundColor).toBe('#303030');
		expect(
			wrapper
				.find('ThemeDemo')
				.find('Text')
				.last()
				.text()
		).toBe('BlueBase Dark');
	});

	test(`should throw render error state if theme is not found`, async () => {
		const BB = new BlueBase();
		await BB.boot({ configs: { 'theme.name': 'foo' } });

		const wrapper = mount(
			<BlueBaseContext.Provider value={BB}>
				<ThemeProvider>
					<ThemeDemo />
				</ThemeProvider>
			</BlueBaseContext.Provider>
		);

		// Wait for render
		await waitForElement(wrapper, 'ErrorState');

		// Verify that background color is dark
		const error = wrapper.find('ErrorState').last();

		expect((error.prop('error') as Error).message).toBe(
			'Could not change theme. Reason: Theme with the key "foo" does not exist.'
		);

		const retry: any = error.prop('retry');

		retry();

		wrapper.unmount();
	});

	test(`should override a theme`, async () => {
		const BB = new BlueBase();
		await BB.boot({ configs: { 'theme.name': 'foo' } });

		const wrapper = mount(
			<BlueBaseContext.Provider value={BB}>
				<ThemeProvider
					theme="bluebase-dark"
					overrides={{ palette: { background: { default: 'red' } } }}
				>
					<ThemeDemo />
				</ThemeProvider>
			</BlueBaseContext.Provider>
		);

		// Wait for render
		await waitForElement(wrapper as any, ThemeDemo);
		// expect(wrapper).toMatchSnapshot();

		// Check theme
		const view = wrapper.find('ThemeDemo [testID="box"]').first();
		expect(view.prop('style')!.backgroundColor).toBe('red');
		expect(
			wrapper
				.find('ThemeDemo Body1 Text')
				.last()
				.text()
		).toBe('BlueBase Dark');

		wrapper.unmount();
	});
});
