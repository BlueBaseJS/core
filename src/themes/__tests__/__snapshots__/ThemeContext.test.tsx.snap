// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContext should change theme based on callback function 1`] = `
<BlueBaseApp>
  <Text>
    <Text>
      Loading
    </Text>
  </Text>
</BlueBaseApp>
`;

exports[`ThemeContext should change theme based on callback function 2`] = `
<BlueBaseApp>
  <BlueBaseRoot>
    <ThemeProvider>
      <ThemedComponent>
        <SystemApp>
          <ThemedComponent>
            <View>
              <View>
                <ThemedComponent>
                  <Component />
                </ThemedComponent>
                <ThemePicker>
                  <Picker
                    mode="dialog"
                    onValueChange={[Function]}
                    selectedValue="bluebase-light"
                    style={
                      Object {
                        "width": 150,
                      }
                    }
                  >
                    <PickerIOS
                      mode="dialog"
                      onValueChange={[Function]}
                      selectedValue="bluebase-light"
                      style={
                        Object {
                          "width": 150,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "width": 150,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "width": 150,
                            }
                          }
                        >
                          <_class
                            items={
                              Array [
                                Object {
                                  "label": "BlueBase Light",
                                  "textColor": undefined,
                                  "value": "bluebase-light",
                                },
                                Object {
                                  "label": "BlueBase Dark",
                                  "textColor": undefined,
                                  "value": "bluebase-dark",
                                },
                              ]
                            }
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            selectedIndex={0}
                            style={
                              Array [
                                Object {
                                  "height": 216,
                                },
                                undefined,
                              ]
                            }
                          >
                            <RCTPicker
                              items={
                                Array [
                                  Object {
                                    "label": "BlueBase Light",
                                    "textColor": undefined,
                                    "value": "bluebase-light",
                                  },
                                  Object {
                                    "label": "BlueBase Dark",
                                    "textColor": undefined,
                                    "value": "bluebase-dark",
                                  },
                                ]
                              }
                              onChange={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              selectedIndex={0}
                              style={
                                Array [
                                  Object {
                                    "height": 216,
                                  },
                                  undefined,
                                ]
                              }
                            />
                          </_class>
                        </View>
                      </View>
                    </PickerIOS>
                  </Picker>
                </ThemePicker>
                <ThemeDemo>
                  <ThemedComponent
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "padding": 8,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fafafa",
                          "padding": 8,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fafafa",
                            "padding": 8,
                          }
                        }
                      >
                        <ThemedComponent>
                          <Component
                            styles={
                              Object {
                                "root": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "Roboto",
                                  "fontSize": 16,
                                  "fontWeight": "400",
                                },
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                    "fontFamily": "Roboto",
                                    "fontSize": 16,
                                    "fontWeight": "400",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "Roboto",
                                      "fontSize": 16,
                                      "fontWeight": "400",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                BlueBase Light
                              </Text>
                            </Text>
                          </Component>
                        </ThemedComponent>
                      </View>
                    </View>
                  </ThemedComponent>
                </ThemeDemo>
                <ThemedComponent>
                  <Component />
                </ThemedComponent>
              </View>
            </View>
          </ThemedComponent>
        </SystemApp>
      </ThemedComponent>
    </ThemeProvider>
  </BlueBaseRoot>
</BlueBaseApp>
`;

exports[`ThemeContext should change theme based on callback function 3`] = `
<BlueBaseApp>
  <BlueBaseRoot>
    <ThemeProvider>
      <ThemedComponent>
        <SystemApp>
          <ThemedComponent>
            <View>
              <View>
                <ThemedComponent>
                  <Component />
                </ThemedComponent>
                <ThemePicker>
                  <Picker
                    mode="dialog"
                    onValueChange={[Function]}
                    selectedValue="bluebase-light"
                    style={
                      Object {
                        "width": 150,
                      }
                    }
                  >
                    <PickerIOS
                      mode="dialog"
                      onValueChange={[Function]}
                      selectedValue="bluebase-light"
                      style={
                        Object {
                          "width": 150,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "width": 150,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "width": 150,
                            }
                          }
                        >
                          <_class
                            items={
                              Array [
                                Object {
                                  "label": "BlueBase Light",
                                  "textColor": undefined,
                                  "value": "bluebase-light",
                                },
                                Object {
                                  "label": "BlueBase Dark",
                                  "textColor": undefined,
                                  "value": "bluebase-dark",
                                },
                              ]
                            }
                            onChange={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            selectedIndex={0}
                            style={
                              Array [
                                Object {
                                  "height": 216,
                                },
                                undefined,
                              ]
                            }
                          >
                            <RCTPicker
                              items={
                                Array [
                                  Object {
                                    "label": "BlueBase Light",
                                    "textColor": undefined,
                                    "value": "bluebase-light",
                                  },
                                  Object {
                                    "label": "BlueBase Dark",
                                    "textColor": undefined,
                                    "value": "bluebase-dark",
                                  },
                                ]
                              }
                              onChange={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              selectedIndex={0}
                              style={
                                Array [
                                  Object {
                                    "height": 216,
                                  },
                                  undefined,
                                ]
                              }
                            />
                          </_class>
                        </View>
                      </View>
                    </PickerIOS>
                  </Picker>
                </ThemePicker>
                <ThemeDemo>
                  <ThemedComponent
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "padding": 8,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#fafafa",
                          "padding": 8,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#fafafa",
                            "padding": 8,
                          }
                        }
                      >
                        <ThemedComponent>
                          <Component
                            styles={
                              Object {
                                "root": Object {
                                  "color": "rgba(0, 0, 0, 0.87)",
                                  "fontFamily": "Roboto",
                                  "fontSize": 16,
                                  "fontWeight": "400",
                                },
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "rgba(0, 0, 0, 0.87)",
                                    "fontFamily": "Roboto",
                                    "fontSize": 16,
                                    "fontWeight": "400",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "fontFamily": "Roboto",
                                      "fontSize": 16,
                                      "fontWeight": "400",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                BlueBase Light
                              </Text>
                            </Text>
                          </Component>
                        </ThemedComponent>
                      </View>
                    </View>
                  </ThemedComponent>
                </ThemeDemo>
                <ThemedComponent>
                  <Component />
                </ThemedComponent>
              </View>
            </View>
          </ThemedComponent>
        </SystemApp>
      </ThemedComponent>
    </ThemeProvider>
  </BlueBaseRoot>
</BlueBaseApp>
`;

exports[`ThemeContext should render a ThemeDemo component with themed background color 1`] = `
<Text>
  Loading
</Text>
`;

exports[`ThemeContext should render a ThemeDemo component with themed background color 2`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#fafafa",
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ]
      }
    >
      BlueBase Light
    </Text>
  </View>
</View>
`;

exports[`ThemeContext should render a overwritten ThemeDemo component with red background color 1`] = `
<Text>
  Loading
</Text>
`;

exports[`ThemeContext should render a overwritten ThemeDemo component with red background color 2`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "400",
          },
          undefined,
        ]
      }
    >
      BlueBase Dark
    </Text>
  </View>
</View>
`;
